src/components/utils/AmplifyApiCall.js

```diff
@@ -2,17 +2,20 @@ import { get } from "aws-amplify/api";
 import { fetchAuthSession } from "aws-amplify/auth"
 
 export class AmplifyApiCall {
-  static async getAuthToken() {
+    static getIdTokenFromLocalStorage() {
+        return localStorage.getItem(Object.keys(localStorage).filter((keyValue) => keyValue.startsWith("CognitoIdentityServiceProvider") && keyValue.endsWith(".idToken"))[0]);
```

Comment:
Would `fetchAuthSession` do this automatically for you? Unless specifying the `forceRefresh` props it will use the same token?

Reply:
yes, the logic here is we can use the idToken generated when we login rather then we fetch it from api call. 
The forceRefresh we will leave it to backend check, if it expired, we should be force logout. Do we have that check of token in backend? or we can use refreshToken as well to realize the token auto-renew
![image](https://github.com/umccr/data-portal-status-page/assets/160200168/2f63c8a0-0a1d-4816-a688-0b4d1eaf4220)

Reply:
I will have a look at the refresh token logic of the api call....

Reply:
> the logic here is we can use the idToken generated when we login rather then we fetch it from api call.

Yes, but by default the `fetchAuthSession` is doing this for you. The token API is only called once and it will keep reusing that token from the local storage (not calling the token APIs until it is needed)

![Screenshot 2024-03-19 at 15 59 19](https://github.com/umccr/data-portal-status-page/assets/61998484/88eb3eb6-b306-4c90-825f-0e56b9fc3369)

Reply:
After testing, the refresh token will be used to auto refresh the id token. I will show you...